design/token.go

```diff
@@ -50,6 +50,45 @@ var _ = a.Resource("token", func() {
 		a.Response(d.BadRequest, JSONAPIErrors)
 		a.Response(d.InternalServerError, JSONAPIErrors)
 	})
+
+	a.Action("link", func() {
+		a.Routing(
+			a.POST("link"),
+		)
+		a.Payload(linkPayload)
+		a.Description("Link the user account to an external resource provider such as GitHub")
+		a.Response(d.SeeOther)
+		a.Response(d.Unauthorized, JSONAPIErrors)
+		a.Response(d.BadRequest, JSONAPIErrors)
+		a.Response(d.InternalServerError, JSONAPIErrors)
+	})
+
+	a.Action("callback", func() {
+		a.Routing(
+			a.GET("/callback"),
+		)
+		a.Params(func() {
+			a.Param("code", d.String, "Code provided by an external oauth2 resource provider")
+			a.Param("state", d.String, "State generated by the link request")
+			a.Required("code")
+			a.Required("state")
+		})
+		a.Description("Callback from an external oauth2 resource provider such as GitHub as part of user's account linking")
+		a.Response(d.TemporaryRedirect)
+		a.Response(d.BadRequest, JSONAPIErrors)
+		a.Response(d.InternalServerError, JSONAPIErrors)
+	})
+})
+
+var linkPayload = a.Type("LinkPayload", func() {
+	a.Attribute("for", d.String, "Resource we need to link accounts for", func() {
+		a.Example("https://github.com/somecoolrepo")
+	})
+	a.Attribute("token", d.String, "User's access token")
+	a.Attribute("redirect", d.String, "URL to be redirected to after successful account linking. If not set then will redirect to the referrer instead.")
+	//a.Attribute("scope", d.String, "Required scope. Multiple scopes can be specified by separating them with a space. If not defined then the default scope is used.")
```

Comment:
My not-so-strong opinion on this is : we can keep the `scope` but make it optional and always default it to our current scopes. :)